inc/saidot1brfdb.h

```diff
@@ -0,0 +1,130 @@
+/*
+* Copyright (c) 2015 Dell Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may
+* not use this file except in compliance with the License. You may obtain
+* a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*
+* THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR
+* CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+* LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+* FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+*
+* See the Apache Version 2.0 License for specific language governing
+* permissions and limitations under the License.
+*
+* @file saidot1brfdb.h
+*
+* @brief This module defines SAI API for IEEE 802.1BR ECID based forwarding
+         functionality. This is applicable only to Port Extenders (PE).
+*************************************************************************************/
+
+#if !defined (__SAIDOT1BRFDB_H)
+#define __SAIDOT1BRFDB_H
+
+#include "saitypes.h"
+#include "saistatus.h"
+
+/** \defgroup SAIDOT1BRFDBENTRY SAI - 802.1BR ECID based forwarding specific public APIs and datastructures in PE.
+ *
+ *  \{
+ */
+
+/**
+ * @brief SAI attributes for SAI_OBJECT_TYPE_DOT1BR_FDB_ENTRY 
+ */
+typedef enum _sai_dot1br_fdb_entry_attr_t
```

Comment:
Dell - Internal Use - Confidential
Hi Atit Jain,

There are few differences between existing FDB object and the proposed dot1brFdb object:

1.      The key (sai_fdb_entry_t) to existing FDB object is <Vlan Id, Mac Addr> tuple, whereas the key to dot1brFdb object is <Ecid>. If we want to merge these 2 objects then sai_fdb_entry_t must be converted to a union.

2.      The dot1brfdb entries cannot be learnt (and hence cannot be aged out too). They can only be configured.

3.      As a result of point 2, there is no need for fdb event notification and flushing of entries

So the object key and most of the attributes of existing FDB are not relevant to dot1brfdb object. I feel it will be better not to merge the 2 objects.

Regards,
Ravi

From: Atit Jain [mailto:notifications@github.com]
Sent: Thursday, June 16, 2016 10:10 PM
To: opencomputeproject/SAI <SAI@noreply.github.com>
Cc: Sivasankar, Ravikumar <Ravikumar_Sivasankar@Dell.com>; Author <author@noreply.github.com>
Subject: Re: [opencomputeproject/SAI] IEEE 802.1BR SAI proposal. (#73)


In inc/saidot1brfdb.h<https://github.com/opencomputeproject/SAI/pull/73#discussion_r67380186>:

> +

> +#if !defined (__SAIDOT1BRFDB_H)

> +#define __SAIDOT1BRFDB_H

> +

> +#include "saitypes.h"

> +#include "saistatus.h"

> +

> +/** \defgroup SAIDOT1BRFDBENTRY SAI - 802.1BR ECID based forwarding specific public APIs and datastructures in PE.

> + *

> + *  \{

> + */

> +

> +/**

> + * @brief SAI attributes for SAI_OBJECT_TYPE_DOT1BR_FDB_ENTRY

> + */

> +typedef enum _sai_dot1br_fdb_entry_attr_t

Cant this be merged with the existing fdb object. Can have an attribute to mark egress as BR port/list or if the BR port list is non null, the FDB points to BR list.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/opencomputeproject/SAI/pull/73/files/4302a4c03dbc322a2d63ce4cc56895a3d50bf984#r67380186>, or mute the thread<https://github.com/notifications/unsubscribe/AOoHu24dSgh9RwsZ4xVmEBjBVIHpnWOOks5qMXxagaJpZM4HpSGE>.